src/DSharp.Mscorlib/Lazy.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace System
+{
+    [ScriptIgnoreGenericArguments]
+    public class Lazy<T>
+    {
+        [ScriptAlias("IsValueCreated")]
+        public extern bool IsValueCreated { get; }
+
+        [ScriptAlias("Value")]
+        public extern T Value { get; }
+
+        public extern Lazy(Func<T> factory);
```

Comment:
Can we add support for `Lazy<T>(T value)` as well, so if we do any unit testing with this code we can supply a fixed value etc. We can support multiple constructors, just inside the js check to see if the parameter is a function or a value.